README.rst

```diff
@@ -1,102 +1,208 @@
-become available.
-
-For a description of one way in which stuck macros might be useful,
-please see this `talk`_.
-
-.. _talk: https://www.youtube.com/watch?v=nUvKoG_V_U0
-
-
-Installation
-============
-
-Klister is written in Haskell, and is built with HPack and Cabal. You can build
-it like so::
-
-    hpack package.yaml
-    cabal v2-build
-
-You can use ``v2-install`` instead of ``v2-build`` to install the Klister
-executable and associated files to your home directory.
-
-Usage
-=====
-
-Run ``klister repl`` to be dropped into a read-eval-print loop where you can try
-out writing some Klister expressions.
-
-``klister run file.kl`` will evaluate the examples and run the IO
-actions in ``file.kl``.
+Klister [`TyDe 2020`_, `video`_] is a research-stage programming language which combines
+the core features of Racket and Haskell into a single language. It is named
+after its most distinguishing feature, "stuck macros" [`Compose NYC 2019`_], as
+"Klister" is Danish for "adhesive".
+
+.. _TyDe 2020: http://davidchristiansen.dk/pubs/tyde2020-predictable-macros-abstract.pdf
+.. _video: https://www.youtube.com/watch?v=FyeWwYfqTHo&t=996s
+.. _Compose NYC 2019: https://www.youtube.com/watch?v=nUvKoG_V_U0
+
+::
+
+  #lang "prelude.kl"
+
+  -- do notation is not builtin syntax, it's implemented as a library!
+  (import "monad.kl")
+
+  -- An effectful action whose inferred type is (-> String (IO Unit))
+  (defun putStrLn (str)
+    (write stdout (string-append str "\n")))
+
+  -- "run" is like main, except you can have more than one.
+  (run
+    -- Klister doesn't have typeclasses yet, so "do" needs an explicit
+    -- dictionary argument.
+    (do io-monad
+      (putStrLn "hello")
+      (putStrLn "world")))
+
+You can run the above program using either stack or cabal::
+
+    $ cabal run klister -- run examples/hello.kl
+    hello
+    world
+
+Features
+========
+
+Features we borrow from Racket:
+
+* `Custom syntax`_, via `hygienic macros`_ with `easy-to-override hygiene`_.
+* `Custom languages`_ (``#lang``), via macros which reinterpret terms into
+  those of an existing ``#lang``.
+* `Syntax objects`_, that is, s-expressions annotated with source locations and
+  lexical information.
+* A `module system`_ which allow imports and exports to be renamed.
+
+.. _Custom syntax: examples/lambda-case.golden
+.. _hygienic macros: TODO: write a short example demonstrating lack of capture.
+.. _easy-to-override hygiene: examples/anaphoric-if.kl
+.. _Custom languages: examples/rpn.kl
+.. _Syntax objects: TODO: link to a short example which explains that in
+   Racket, syntax objects are introduced via ``#'(...)``, whereas in Klister
+   they are introduced via ``'(...)``. Also explain that Klister does not have
+   unannotated s-expressions.
+.. _module system: TODO: write a short example demonstrating what can be
+   expressed with the import and export primitives.
+
+Features we borrow from Haskell:
```

Comment:
```suggestion
Features we borrow from the ML family of languages and Haskell:
```

Reply:
How about this into two separate sections: we borrow the type system from ML, and the rest from Haskell?

Reply:
I tried my counter-suggestion in 9232b80a847b8c3b51fceeca5bca6c6ab48eeabc, let me know what you think!

Reply:
Looks good!